src/app/clusters/service-area-server/service-area-server.cpp

```diff
@@ -0,0 +1,1104 @@
+    });
+}
+
+//*************************************************************************
+// command handlers
+
+void Instance::HandleSelectLocationsCmd(HandlerContext & ctx, const Commands::SelectLocations::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "Service Area: HandleSelectLocationsCmd");
+
+    uint32_t listIndex = 0;
+    uint32_t oldSelectedLocation;
+    size_t numberOfLocations = 0;
+    bool matchesCurrentSelectedLocations;
+
+    // On receipt of this command the device SHALL respond with a SelectLocationsResponse command.
+    auto exitResponse = [ctx](SelectLocationsStatus status, CharSpan statusText) {
+        Commands::SelectLocationsResponse::Type response{
+            .status     = status,
+            .statusText = Optional(statusText),
+        };
+        ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+    };
+
+    // Get the number of Selected Locations in the command parameter and check that it is valid.
+    if (!req.newLocations.IsNull())
+    {
+        if (CHIP_NO_ERROR != req.newLocations.Value().ComputeSize(&numberOfLocations))
+        {
+            exitResponse(SelectLocationsStatus::kInvalidSet, "decoding of newLocations parameter size failed"_span);
+            return;
+        }
+
+        // If the device determines that it can't operate at all locations from the list,
+        // the SelectLocationsResponse command's Status field SHALL indicate InvalidSet.
+        if (((numberOfLocations == 0) || (numberOfLocations > kMaxNumSelectedLocations)))
+        {
+            exitResponse(SelectLocationsStatus::kInvalidSet, "invalid number of locations"_span);
+            return;
+        }
+    }
+
+    // if number of selected locations in parameter matches number in attribute - the locations *might* be the same
+    matchesCurrentSelectedLocations = (numberOfLocations == mDelegate->GetNumberOfSelectedLocations());
+
+    if (!req.newLocations.IsNull())
+    {
+        // do as much parameter validation as we can
+        {
+            uint32_t i         = 0;
+            auto iLocationIter = req.newLocations.Value().begin();
+            while (iLocationIter.Next())
+            {
+                uint32_t aSelectedLocation = iLocationIter.GetValue();
+
+                // each item in this list SHALL match the LocationID field of an entry on the SupportedLocations attribute's list
+                // If the Status field is set to UnsupportedLocation, the StatusText field SHALL be an empty string.
+                if (!IsSupportedLocation(aSelectedLocation))
+                {
+                    exitResponse(SelectLocationsStatus::kUnsupportedLocation, ""_span);
+                    return;
+                }
+
+                // Checking for duplicate locations.
+                uint32_t j         = 0;
+                auto jLocationIter = req.newLocations.Value().begin();
+                while (j < i)
+                {
+                    jLocationIter.Next();
```

Comment:
Document why it's OK to not check the return value here (which I agree it is).

Reply:
Added [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/efafc7c11f9c4920a385ab2956b662aaa73351d3).